geopolars/src/geoseries.rs

```diff
@@ -503,35 +551,46 @@ impl GeoSeries for Series {
     }
 
-    fn rotate(&self, angle: f64, origin: TransformOrigin) -> Result<Series> {
+    fn rotate<'a>(
+        &self,
+        angle: impl Into<BroadcastableParameter<'a, f64>>,
+        origin: TransformOrigin,
+    ) -> Result<Series> {
         use geo::algorithm::bounding_rect::BoundingRect;
         use geo::algorithm::centroid::Centroid;
-        match origin {
-            TransformOrigin::Centroid => {
-                let rotated_geoms: Vec<Geometry<f64>> = iter_geom(self)
-                    .map(|geom| {
-                        let centroid = geom.centroid().unwrap();
-                        let transform = AffineTransform::rotate(angle, centroid);
-                        geom.map_coords(|c| transform.apply(c))
-                    })
-                    .collect();
-                Series::from_geom_vec(&rotated_geoms)
-            }
-            TransformOrigin::Center => {
-                let rotated_geoms: Vec<Geometry<f64>> = iter_geom(self)
-                    .map(|geom| {
-                        let center = geom.bounding_rect().unwrap().center();
-                        let transform = AffineTransform::rotate(angle, center.into());
-                        geom.map_coords(|c| transform.apply(c))
-                    })
-                    .collect();
-                Series::from_geom_vec(&rotated_geoms)
-            }
-            TransformOrigin::Point(point) => {
-                let transform = AffineTransform::rotate(angle, point);
-                self.affine_transform(transform)
-            }
-        }
+
+        let apply_rotation: Box<dyn Fn(&Geometry<f64>, f64) -> Geometry<f64>> = match origin {
+            TransformOrigin::Centroid => Box::new(|geom: &Geometry<f64>, angle: f64| {
+                let centroid = geom.centroid().unwrap();
+                let transform = AffineTransform::rotate(angle, centroid);
+                geom.map_coords(|c| transform.apply(c))
+            }),
+            TransformOrigin::Center => Box::new(|geom: &Geometry, angle: f64| {
+                let center = geom.bounding_rect().unwrap().center();
+                let transform = AffineTransform::rotate(angle, center.into());
+                geom.map_coords(|c| transform.apply(c))
+            }),
+            TransformOrigin::Point(point) => Box::new({
+                move |geom: &Geometry, angle: f64| {
+                    let transform = AffineTransform::rotate(angle, point);
+                    geom.map_coords(|c| transform.apply(c))
+                }
+            }),
+        };
+
+        let angle: BroadcastableParameter<'a, f64> = angle.into();
+
+        let rotated_geoms: Vec<Geometry<f64>> = iter_geom(self)
+            .enumerate()
+            .map(|(index, geom)| match angle {
+                BroadcastableParameter::Val(v) => apply_rotation(&geom, v),
+                BroadcastableParameter::Series(s) => {
+                    apply_rotation(&geom, s.f64().unwrap().get(index).unwrap())
```

Comment:
One question here: if we're now taking `series` as input, then how should we handle its inherent dynamic typing? Here the code would panic if the series contained any data type other than `f64`. What if the series contained `f32`...? Seems like a situation where we'd want several numeric data types to work seamlessly